src/main/java/ca/ryangreen/apigateway/generic/GenericApiGatewayRequest.java

```diff
@@ -4,20 +4,25 @@
 
 import java.io.IOException;
 import java.io.InputStream;
+import java.util.List;
 import java.util.Map;
 
 public class GenericApiGatewayRequest {
+
     private final HttpMethodName httpMethod;
     private final String resourcePath;
     private final InputStream body;
     private final Map<String, String> headers;
+    private final Map<String, List<String>> parameters;
```

Comment:
Should this be called queryStringParameters?

Reply:
parameters seemed to be in the same flavor of 'headers', 'body' etc.  Also it is setParameters() in the  DefaultRequest